tests/update_workflow_test.go

```diff
@@ -4684,55 +4692,42 @@ func (s *functionalSuite) TestUpdateWorkflow_StaleSpeculativeWorkflowTask_ClearM
 	})
 	s.IsType(&serviceerror.NotFound{}, err)
 
-	// Complete of the 4th WT should succeed
+	// Complete of the 4th WT should succeed. It accepts both updates.
 	wt5Resp, err := s.engine.RespondWorkflowTaskCompleted(testCtx, &workflowservice.RespondWorkflowTaskCompletedRequest{
 		Namespace: s.namespace,
 		TaskToken: wt4.TaskToken,
 		Commands: append(
-			s.acceptUpdateCommands(tv, "2"),
-			&commandpb.Command{
-				CommandType: enumspb.COMMAND_TYPE_SCHEDULE_ACTIVITY_TASK,
-				Attributes: &commandpb.Command_ScheduleActivityTaskCommandAttributes{ScheduleActivityTaskCommandAttributes: &commandpb.ScheduleActivityTaskCommandAttributes{
-					ActivityId:             tv.ActivityID("3"),
-					ActivityType:           tv.ActivityType(),
-					TaskQueue:              tv.TaskQueue(),
-					ScheduleToCloseTimeout: tv.InfiniteTimeout(),
-				}},
-			}),
-		Messages:              s.acceptUpdateMessages(tv, wt4.Messages[0], "2"),
+			s.acceptUpdateCommands(tv, "1"),
```

Comment:
Just FYI: you need that many nested appends because go is missing `flatten`. Flatten is a basic function which receives a slice of slices and returns one slice with all the elements from all the slices.